data/caasp/update.sh

```diff
@@ -131,18 +131,21 @@ elif [ ! -z "${INSTALL:-}" ]; then
             $runner "zypper lr UPDATE | grep Enabled"
 
             # Install the packages
-            $runner "/usr/sbin/transactional-update pkg install -y $(cat not_installed.txt)"
+            $runner "/usr/sbin/transactional-update salt pkg install -y $(cat not_installed.txt)"
+
+	    # Refresh the Grains
+	    $srun '*' saltutil.refresh_grains
 
             # Enable the update repo
             $runner "zypper mr -e UPDATE"
             $runner "zypper lr UPDATE | grep Enabled"
 
-	    # Reboot the cluster
-            where="-P roles:(admin|kube-(master|minion))"
-	    srun="docker exec -d $saltid salt"
-	    $srun $where system.reboot
+            echo 'QAM_INSTALL'
+	    # Orchestrate the reboot via Velum to complete the installation
```

Comment:
"Orchestrate the reboot via Velum" is missing, or? What I think happens is:
```
0. current state (active subvolume = 0, default sv =0)

# Install package - transactional-update creates new subvolume (with installed package)
# and set it as default, so it will be activated after reboot
1. ./update.sh -i (active sv = 0, default sv = 1)

# Update package - transactional-update will try to update ACTIVE subvolume (0)
# which does not contain your installed package. There is nothing to update!
2. ./update.sh -u (active sv = 0, default sv = 1)

# You apply the transactional-update changes
3. reboot via Velum (active sv = 1, default sv =1)
```
So it's like step 2. never happened.
You need to do is reboot after transactional-update  command to apply the changes

Reply:
What was happening is that in step 1:

```
- $runner "/usr/sbin/transactional-update pkg install -y $(cat not_installed.txt)"
+ $runner "/usr/sbin/transactional-update salt pkg install -y $(cat not_installed.txt)"
```

The `salt` parameter was missing. This tells velum via *salt grains*, that a reboot is required. So now test goes into Velum and orchestrates the reboot from there and everything is fine.

Before this PR, since the `salt` parameter was missing, I was manually rebooting the cluster using `$srun $where system.reboot` called from the `update.sh` script. It seems that CaaSP **doesn't like that**:

> As of now there's no any mechanism in CaaSP to ensure that your cluster will come back in good state in case of cold reboot or shutdown that wasn't handled by Velum. e.g. in a situation of power-outage (so this is a new bug which I will file next week) the cluster cannot be restored in a good state.

In step2, the code for the reboot was/is in openQA `stack_update`, not in the *bash* script. So the reboot was happening properly in this -- *step 2* -- case. But ...  if there was another reboot before that (the one outside of Velum, in case a package needed to be installed first) then the cluster wasn't always healthy, so the orchestration of part 2 was failing (for the wrong reasons) because the reboot of part1 didn't happen properly. 

Anyway, now both steps (1 and 2) makes sure that the reboot it not handled by the script `update.sh` but by Velum, and everything works fine.

Reply:
You can try this manually on MicroOS:
```
transactional-udpate pkg install xxx;
transactional-udpate pkg install yyy;
reboot
```
Only package `yyy` is present on the system after reboot